static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js

```diff
@@ -174,18 +174,18 @@ const makeABTestVariant = (
     parentTest: EpicABTest
 ): Variant => {
     const trackingCampaignId = `epic_${parentTest.campaignId}`;
-    const campaignCode = getCampaignCode(
-        parentTest.campaignPrefix,
-        parentTest.campaignId,
-        id,
-        parentTest.campaignSuffix
-    );
     const iframeId = `${parentTest.campaignId}_iframe`;
 
     // defaults for options
     const {
         maxViews = defaultMaxViews,
         isUnlimited = false,
+        campaignCode = getCampaignCode(
```

Comment:
I've added the ability for a variants to specify a custom `campaignCode` which will override the normal logic which adds a generic prefix and a suffix based on the variant name.

I don't think we need this logic at all anymore since we have an explicit `abTest` field and we don't analyse tests using the campaignCode. But I want to check with the team and simplify this in a separate PR

Reply:
#18273 can be closed in preference for this. As discussed with @joelochlann, a discussion is needed to first check we are ok to stop including campaign codes in components that aren't part of a campaign.